build.gradle

```diff
@@ -354,6 +354,30 @@ project(':iceberg-aws') {
   }
 }
 
+project(':iceberg-gcp') {
+  dependencies {
+    implementation project(path: ':iceberg-bundled-guava', configuration: 'shadow')
+    api project(':iceberg-api')
+    implementation project(':iceberg-common')
+    implementation project(':iceberg-core')
+
+    implementation platform('com.google.cloud:libraries-bom:24.1.0')
+    implementation 'com.google.cloud:google-cloud-storage'
+
+    testImplementation 'com.google.cloud:google-cloud-nio'
+    testImplementation 'org.assertj:assertj-core'
```

Comment:
Is it possible to add this as a testImplementation to all sub-projects (could be a follow up)?

Reply:
There was a discussion about this perviously and I'm not sure where it landed, but some subprojects are using it now while others are not.

Reply:
I'm fine with it.

Reply:
Let's keep that as a follow up since it's a small change and easier to review separately.